### PR TITLE
Fix raw number operations of different functions to use Biggystring

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

### CODE REVIEW

File: src/components/common/ExchangeRate.js [hunk 1]
```diff
@@ -33,11 +33,11 @@ class ExchangeRateComponent extends React.Component<Props & ThemeProps> {
     const primaryDisplayName: string = primaryInfo.displayDenomination.name
     const secondaryDisplaySymbol: string = secondaryInfo.displayDenomination.symbol
     const getDisplayExchangeAmount = secondaryDisplayAmount => {
-      const primaryRatio = parseInt(primaryInfo.displayDenomination.multiplier) / parseInt(primaryInfo.exchangeDenomination.multiplier)
-      const secondaryRatio = parseInt(secondaryInfo.displayDenomination.multiplier) / parseInt(secondaryInfo.exchangeDenomination.multiplier)
-      return (primaryRatio / secondaryRatio) * parseFloat(secondaryDisplayAmount)
+      const primaryRatio = bns.div(primaryInfo.displayDenomination.multiplier, primaryInfo.exchangeDenomination.multiplier)
+      const secondaryRatio = bns.div(secondaryInfo.displayDenomination.multiplier, secondaryInfo.exchangeDenomination.multiplier)
+      return bns.mul(bns.div(primaryRatio, secondaryRatio), secondaryDisplayAmount.toString())
```
Comment (thehobbit85) [inline src/components/common/ExchangeRate.js:38]: I think 'secondaryDisplayAmount' is already a string no?

File: src/components/common/ExchangeRate.js [hunk 2]
```diff
@@ -33,11 +33,11 @@ class ExchangeRateComponent extends React.Component<Props & ThemeProps> {
...
     }
-    let precision = secondaryInfo.displayDenomination.multiplier ? log10(secondaryInfo.displayDenomination.multiplier) : 0
+    let precision = secondaryInfo.displayDenomination.multiplier ? bns.log10(secondaryInfo.displayDenomination.multiplier) : 0
```
Comment (thehobbit85) [inline src/components/common/ExchangeRate.js:40]: Isn't 'secondaryInfo.displayDenomination.multiplier' a number and not a string?

File: src/components/scenes/TransactionDetailsScene.js [hunk 3]
```diff
@@ -410,8 +410,9 @@ export class TransactionDetailsComponent extends React.Component<Props, State> {
 
     const amount = currentFiatAmount ? bns.toFixed(currentFiatAmount, 2, 2) : '0'
     const fiatAmount = amountFiat.replace(',', '.')
-    const difference = amount ? parseFloat(amount) - parseFloat(fiatAmount) : 0
-    const percentageFloat = amount && parseFloat(fiatAmount) > 0 ? (difference / parseFloat(fiatAmount)) * 100 : 0
+    const difference = amount ? bns.sub(amount, fiatAmount) : '0'
+    // Cannot us bns.div because div Truncates remainder
+    const percentageFloat = amount && parseFloat(fiatAmount) > 0 ? bns.mul((parseFloat(difference) / parseFloat(fiatAmount)).toString(), '100') : 0
```
Comment (thehobbit85) [inline src/components/scenes/TransactionDetailsScene.js:415]: Don't use "parseFloat", it kinda ruins the whole point to not use number, just use bns for everything.

File: src/components/themed/WalletListCurrencyRow.js [hunk 4]
```diff
@@ -67,14 +74,14 @@ class WalletListRowComponent extends React.PureComponent<Props> {
...
-    } else if (exchangeRate && differencePercentage && differencePercentage < 0) {
+    } else if (exchangeRate && differencePercentage && parseFloat(differencePercentage) < 0) {
       differencePercentageStyle = 'Negative'
-      differencePercentageString = `-${Math.abs(differencePercentage).toFixed(1)}%`
-    } else if (exchangeRate && differencePercentage && differencePercentage > 0) {
+      differencePercentageString = `-${Math.abs(parseFloat(differencePercentage)).toFixed(1)}%`
+    } else if (exchangeRate && differencePercentage && parseFloat(differencePercentage) > 0) {
       differencePercentageStyle = 'Positive'
-      differencePercentageString = `+${Math.abs(differencePercentage).toFixed(1)}%`
+      differencePercentageString = `+${Math.abs(parseFloat(differencePercentage)).toFixed(1)}%`
```
Comment (thehobbit85) [inline src/components/themed/WalletListCurrencyRow.js:84]: No need to go back to numbers for comparisons, you should use this:

https://www.npmjs.com/package/biggystring#comparisons

File: src/components/themed/WalletListCurrencyRow.js [hunk 5]
```diff
@@ -54,10 +55,16 @@ class WalletListRowComponent extends React.PureComponent<Props> {
...
-    const yesterdayExchangeRate = yesterdayUsdExchangeRate * fiatExchangeRate
-    const differenceYesterday = exchangeRate ? exchangeRate - yesterdayExchangeRate : null
+    const yesterdayExchangeRate =
+      yesterdayUsdExchangeRate != null && fiatExchangeRate != null ? bns.mul(yesterdayUsdExchangeRate.toString(), fiatExchangeRate.toString()) : '0'
+
+    const differenceYesterday = exchangeRate ? bns.sub(exchangeRate.toString(), yesterdayExchangeRate) : null
+
+    let differencePercentage =
+      differenceYesterday && yesterdayExchangeRate !== '0'
+        ? bns.mul((parseFloat(differenceYesterday) / parseFloat(yesterdayExchangeRate)).toString(), '100')
```
Comment (thehobbit85) [inline src/components/themed/WalletListCurrencyRow.js:65]: Don't use parseFloat, just stay in strings and use bns

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments

File: src/components/themed/WalletListCurrencyRow.js (hunk 5 above)
Comment (amanalili) [reply src/components/themed/WalletListCurrencyRow.js:65]: Just multiply by a 100 x 10^(num of decimal places)

**Comment by amanalili** [comment]
Split into separate PR for each function

**Pull request closed without merging**